### PR TITLE
Remove start/end time from return Parameters after initial job creation

### PR DESCRIPTION
**Pull request by feordin**
When a reindex job is first created, it is neither started, nor finished, only queued, so removing start/end time from the returned Parameters resource as the default values where incorrect.  This PR also updates the GET for the reindex job, so you can query the Parameters resource for a job.

## Related issues
Addresses [issue [AB#74594](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/74594)].

## Testing
Additional unit tests added, and manually tested

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs [hunk 1]
```diff
@@ -240,9 +240,18 @@ public Uri ResolveOperationResultUrl(string operationName, string id)
             EnsureArg.IsNotNullOrWhiteSpace(operationName, nameof(operationName));
             EnsureArg.IsNotNullOrWhiteSpace(id, nameof(id));
 
-            if (!string.Equals(operationName, OperationsConstants.Export, StringComparison.OrdinalIgnoreCase))
+            string routName = null;
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs:243]: nit: typo, routeName

File: src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs [hunk 2]
```diff
@@ -240,9 +240,18 @@ public Uri ResolveOperationResultUrl(string operationName, string id)
...
+
+            switch (operationName)
```
Comment (namadabu) [inline src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs:245]: Corresponding UTs for this change? We have some in `UrlResolverTests.cs`

Comment (feordin) [reply src/Microsoft.Health.Fhir.Api/Features/Routing/UrlResolver.cs:245]: Unit tests added.

File: src/Microsoft.Health.Fhir.Api/Features/Headers/ResourceActionResultExtensions.cs [hunk 3]
```diff
@@ -10,27 +10,27 @@
 
 namespace Microsoft.Health.Fhir.Api.Features.Headers
 {
-    public static class ExportResultExtensions
+    public static class ResourceActionResultExtensions
     {
         // Generates the url to be included in the response based on the operation and sets the content location header.
-        public static ExportResult SetContentLocationHeader(this ExportResult exportResult, IUrlResolver urlResolver, string operationName, string id)
+        public static ResourceActionResult<TResult> SetContentLocationHeader<TResult>(this ResourceActionResult<TResult> result, IUrlResolver urlResolver, string operationName, string id)
         {
```
Comment (rbans96) [inline src/Microsoft.Health.Fhir.Api/Features/Headers/ResourceActionResultExtensions.cs:17]: Should we verify result EnsureArg.IsNotNull(result, nameof(result)); ?

File: src/Microsoft.Health.Fhir.Api/Features/Headers/ResourceActionResultExtensions.cs [hunk 4]
```diff
@@ -10,27 +10,27 @@
...
             var url = urlResolver.ResolveOperationResultUrl(operationName, id);
 
-            exportResult.Headers.Add(HeaderNames.ContentLocation, url.ToString());
-            return exportResult;
+            result.Headers.Add(HeaderNames.ContentLocation, url.ToString());
+            return result;
         }
 
-        public static ExportResult SetContentTypeHeader(this ExportResult exportResult, string contentTypeValue)
+        public static ResourceActionResult<TResult> SetContentTypeHeader<TResult>(this ResourceActionResult<TResult> result, string contentTypeValue)
```
Comment (rbans96) [inline src/Microsoft.Health.Fhir.Api/Features/Headers/ResourceActionResultExtensions.cs:28]: Same here?

File: src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Routing/UrlResolverTests.cs [hunk 5]
```diff
@@ -284,6 +284,22 @@ public void GivenAnExportOperation_WhenOperationResultUrlIsResolved_ThenCorrectU
...
+            ValidateUrlRouteContext(
+                RouteNames.GetReindexStatusById,
+                routeValues =>
+                {
+                    Assert.Equal(id, routeValues[KnownActionParameterNames.Id]);
+                });
+        }
+
         [Fact]
         public void GivenANonExportOperation_WhenOperationResultUrlIsResolved_ThenOperationNotImplementedExceptionShouldBeThrown()
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Routing/UrlResolverTests.cs:304]: Could you update the name of this test to reflect the new behavior?

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs [hunk 6]
```diff
@@ -347,9 +347,35 @@ public async Task<bool> CheckActiveReindexJobsAsync(CancellationToken cancellati
             }
         }
 
-        public Task<ReindexJobWrapper> GetReindexJobByIdAsync(string jobId, CancellationToken cancellationToken)
+        public async Task<ReindexJobWrapper> GetReindexJobByIdAsync(string jobId, CancellationToken cancellationToken)
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs:350]: Nit: On line 294 there is a typo, an/a. For some reason GitHub won't let me put a comment there,

**Comment by LTA-Thinking** [comment]
Why does the reindex job have a start time? The export jobs have a queued time, but not a start time.

**Comment by feordin** [comment]
@LTA-Thinking Good point on the Queued time.  The Reindex job does capture the queued time, but I was not reporting it in the output.  I also plan to capture start time when the job begins reindexing.

**Pull request merged**